diff --git a/man/help2man b/man/help2man
index fe786ac..f001c6d 100755
--- a/man/help2man
+++ b/man/help2man
@@ -720,7 +720,7 @@ sub get_option_value
     my $value = join '',
 	map { s/ +$//; expand $_ }
 	map { dec $_ }
-	`$prog $opt 2>$stderr`;
+	`qemu-arm $prog $opt 2>$stderr`;
 
     unless ($value)
     {
diff --git a/src/fs.h b/src/fs.h
index e1034f4..66a795a 100644
--- a/src/fs.h
+++ b/src/fs.h
@@ -2,6 +2,8 @@
    Please send additions to bug-coreutils@gnu.org and meskes@debian.org.
    This file is generated automatically from ./src/stat.c. */
 
+#define __linux__
+
 #if defined __linux__
 # define S_MAGIC_ADFS 0xADF5
 # define S_MAGIC_AFFS 0xADFF
@@ -110,3 +112,5 @@
 #elif defined __GNU__
 # include <hurd/hurd_types.h>
 #endif
+
+#undef __linux__
diff --git a/src/ioblksize.h b/src/ioblksize.h
index 1b91995..21964a3 100644
--- a/src/ioblksize.h
+++ b/src/ioblksize.h
@@ -70,7 +70,7 @@
    In the future we could use the above method if available
    and default to io_blksize() if not.
  */
-enum { IO_BUFSIZE = 128*1024 };
+enum { IO_BUFSIZE = 4*1024 };
 static inline size_t
 io_blksize (struct stat sb)
 {
diff --git a/src/sync.c b/src/sync.c
index 5e1dbb8..3952311 100644
--- a/src/sync.c
+++ b/src/sync.c
@@ -33,6 +33,8 @@
   proper_name ("Jim Meyering"),                 \
   proper_name ("Giuseppe Scrivano")
 
+#undef HAVE_SYNCFS
+
 #ifndef HAVE_SYNCFS
 # define HAVE_SYNCFS 0
 #endif
diff --git a/src/tac.c b/src/tac.c
index 57e7e76..5b33fc8 100644
--- a/src/tac.c
+++ b/src/tac.c
@@ -66,6 +66,7 @@ tac -r -s '.\|
 # define DONT_UNLINK_WHILE_OPEN 1
 #endif
 
+#define DONT_UNLINK_WHILE_OPEN 1
 
 #ifndef DEFAULT_TMPDIR
 # define DEFAULT_TMPDIR "/tmp"
